mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -636,4 +638,81 @@ private ArtifactRepository getArtifactRepository(String runId) {
     URI baseArtifactUri = URI.create(getRun(runId).getInfo().getArtifactUri());
     return artifactRepositoryFactory.getArtifactRepository(baseArtifactUri, runId);
   }
+
+
+  // ********************
+  // * Model Repository *
+  // ********************
+
+  /**
+   * Return the latest model version for each stage (devo, staging, prod).
+   *
+   *    <pre>
+   *        List<ModelVersionDetailed> detailsList = getLatestModelVersionPerStage("model");
+   *
+   *        for (ModelVersionDetailed details : detailsList) {
+   *            System.out.println("Model Name: " + details.getModelVersion().getRegisteredModel().getName());
+   *            System.out.println("Model Version: " + details.getModelVersion().getVersion());
+   *            System.out.println("Current Stage: " + details.getCurrentStage());
+   *        }
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @return A collection of {@link org.mlflow.api.proto.ModelRegistry.ModelVersionDetailed}
+   */
+  public List<ModelVersionDetailed> getLatestModelVersionPerStage(String modelName) {
+    throw new UnsupportedOperationException("getLatestVersion is not currently supported");
+  }
+
+  /**
+   * Return the model URI containing for the given model version. THe model URI can be used to download
+   * the model version artifacts.
+   *
+   * For example, the model URI
+   *
+   *    <pre>
+   *        String modelUri = getModelVersionDownloadUri("model", 0);
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @param modelVersion The version number of the model
+   * @return The specified model version's URI.
+   */
+  public String getModelVersionDownloadUri(String modelName, long modelVersion) {
```

Comment:
`modelVersion` -> `version` for consistency and also because we've been using `version` to refer to the number and `modelVersion` to refer to objects in the rest of MLflow.

Reply:
And `modelName` -> `name`

Reply:
Done.